File: frontend-sist-horto/src/types/DadosMudaResumo.ts
import type { CategoriaMuda } from "./CategoriaMuda";

export interface DadosMudaResumo {
  id: number;
  nomesPopulares: string[];
  nomeCientifico: string;
  categoria: CategoriaMuda;
  familia: string;
  linkImagemArvore: string;
}

File: backend-sist-horto/src/main/java/com/bsh/backend_sist_horto/gestao_mudas/dto/DadosMudaResumo.java
package com.bsh.backend_sist_horto.gestao_mudas.dto;

import com.bsh.backend_sist_horto.gestao_mudas.enums.CategoriaMuda;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;

import java.util.List;

@AllArgsConstructor
@Getter @Setter
public class DadosMudaResumo {
    private Long id;
    private List<String> nomesPopulares;
    private CategoriaMuda categoria;
    private String familia;
    private String linkImagemArvore;
    private Integer estoqueDisponivel;
}